omod/src/main/java/org/openmrs/module/owa/activator/OwaActivator.java

```diff
@@ -163,4 +170,27 @@ public void stopped() {
 		log.info("OWA Module stopped");
 	}
 	
+	/**
+	 * Used to set the login Url Global Property that directs to the Add-on OWA UI if no UI module
+	 * is installed. Searches against the legacy and Reference Application module on installed UI
+	 * modules.
+	 */
+	private void configureLoginUrl(AppManager appManager) {
+		if (uiModuleInstalled() || !appManager.exists(ADD_ON_MANAGER_APP)) {
+			Context.getAdministrationService().setGlobalProperty(LOGIN_URL, "login.htm");
```

Comment:
What is the value of setting this global property when we are returning?

Reply:
None of the ui modules will set this property yet, so if they are installed already, then the default login.url should be reset here to the default.

Reply:
That login.url GP is set once by the core on the first installation.

Reply:
What happens if this value is not set and we just return?

Reply:
Since non of the modules sets this GP and that the owa filter depends on this GP to determine whether it's time to login with the addonmanager, it may always be the addonmanager login page that appears.

Reply:
Are you sure? Did you test and confirm so?

Reply:
Yes @dkayiwa, I'm sure about this.

Reply:
What are the steps to reproduce that problem if i commented out this line?

Reply:
1) Start the platform having the three bundled modules including the owa module with the un-commented line. Confirm that it's the addonmanager login page displaying.

2) Restart the platform having the three bundled modules + legacyui, the included owa module should have the line commented. Confirm that it's still the addonmanager login page displaying.

The launch url could be any of '/', '/index.htm' or '/login.htm'

Reply:
I mean steps to reproduce when the line is removed through all the steps.

Reply:
This means that the login.url GP remains untouched since the last save.

Reply:
As an end user, who is not changing code, if i had the module compiled with that line removed, what steps should i take to reproduce the problem which that line intends to fix?

Reply:
Ohh, okay. The end user starts the platform, and uploads the legacyui. Relaunch the web app in a different tab using /, /login.htm or /index.htm. the landing page remains to be that of the addonmanager.

Reply:
Just tested this and confirmed that you are right. 👍 

Now this is the remaining one:
Installed the platform with owa, fhir and webservices rest.
Got the addon manager screen and logged in.
Then installed the legacyui module using the addon manager which was successful
On going back to the home page, i got the old platform is running page.

Reply:
Quite strange for me, not sure which iteration of the OWA module you are using.

Reply:
Does this happen with the line above uncommented?

Reply:
It happens when that line is in place. (Not commented out)

Reply:
Seems like the legacyui was installed but somehow not started.

Reply:
I checked the database and it had legacyui.started = true

Reply:
That should happen if [this](https://github.com/openmrs/openmrs-module-owa/pull/70/files/64c627f893b316ddbc9f6bb08f6d427aed245128#diff-b8cc126254f819614d8e0a26932db8aeR61) is not there.

Reply:
What i mean is that am just testing your changes as they are in this pull request.

Reply:
Cool @dkayiwa. This is not reproducible on my end, may you give it one more try to confirm the behavior? :-)

Reply:
What are the exact steps that you did?

Reply:
I've done this using the SDK and tomcat. With SDK, 
-I install the platform
-install latest versions of the bundled modules (owa module is this snapshot)
-install addonmanager 1.1.0
-Run/start the platform, access and navigate ui from /, /index.htm and /login.htm.

Reply:
Can you take a second look at my steps?

Reply:
I see, I have followed the very same steps and that's not reproducible for me.

Looks like it's the environments differing probably. I'm using windows with tomcat or SDK.

Reply:
What exact steps have you done?

Reply:
Started the platform with owa, fhir and webservices rest.
Got the addon manager screen and logged in.
Then installed the legacyui module using the addon manager which was successful
On going back to the home page, I don't get the old platform is running page.

Reply:
What happens when you use jetty? mvn jetty:run

Reply:
This returns error `No plugin found for prefix 'jetty'`

Reply:
I recall using this sometime back, somehow I lost its bearings.

Reply:
That happens when you do not run from the webapp folder

Reply:
Ohh yeah thank you. I must have been tired. I was also missing out on some point that this happens within the openmrs-core project rather than the openmrs-distro-platform project.

Reply:
I now appreciate jetty. Too fast. Now learning how that you added the modules, got it from the logs.

Reply:
But on windows my the modules never start with such error,
`Could not start some modules because they depend on others which are not started. Please first start them or use the "Start All" button on the "Manage Modules" page: [fhir->legacyui, owa->legacyui, webservices.rest->legacyui, owa->webservices.rest]`

Reply:
I'm I missing a certain config piece to add to the command `mvn jetty:run` ?

Reply:
No. That should be it. Can you start them all?

Reply:
I'm curious to know how you start the modules manually, especially with jetty. :)

Reply:
jetty just runs the platform. It is the platform which is supposed to run the modules.

Reply:
I see, you are correct that the error exists. I reproduced the issue with Jetty with both the changes here and without any of the changes made here. The same error occurs. 

To add, no legacyui view/page can be reached. I noticed while uploading the the legacyiu that a stack trace showed up to do with Web Component Registrations for the legacyui servelets. So somehow with Jetty, the legacyui module is in an ambiguous started state. Same thing happens if the legacyui module is also placed at once with the other module in the bundledModules folder and jetty run. Except that here, no stacktrace is produced when legacyui module is being started.

This could be a bug between legacyui and jetty. This is random though. There are times when this is working fine, but for the most the error shows up.

Reply:
Gona have to report this on Jira.